### PR TITLE
[PR] Basic individual spacings in elkg and elkt

### PR DESCRIPTION
**Pull request by uruuru**
Light version of #629, addressing #620: allows to specify the individual spacings via a plain string. I.e. no formatting and content assist within `elkt`. But better than nothing 🙄. 

```
spacing.portPort: 5

node n1 {
    spacing.individual: "spacing.portPort: 10"   
    port p1
    port p2 
}

node n2 {
    port p1
    port p2 
}
```

### CODE REVIEW

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/Core.melk [hunk 1]
```diff
@@ -384,12 +384,24 @@ group spacing {
...
-            the children include ports, edges, and labels. "
+    advanced option individual: IndividualSpacings {
+        label "Individual Spacing"
+        description 
+            "Allows to specify individual spacing values for graph elements that shall be different from 
+            the value specified for the element's parent."
+        documentation
+            "In most cases, spacing values apply to the children of the hierarchical node 
+            (possibly the root node) for which the values are actually specified. 
+            Hereby, the children may include ports, edges, and labels.
```
Comment (le-cds) [inline plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/Core.melk:395]: This particular sentence might be misleading. To affect the spacing of node labels, an individual override has to be set on the node. Specifying a node label spacing on the node would not work. Or am I confused?

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/Core.melk [hunk 2]
```diff
@@ -384,12 +384,24 @@ group spacing {
...
+            This option allows to specify a different spacing for a certain element, 
+            thus overriding the parent's value. To give an example, a spacing.portPort of 5 configured 
+            on the graph would result in the ports of all of the graph's nodes to be spaced apart by 5px. 
+            A single node may desire a larger spacing and hence set an individual spacing.portPort of 10.
+            Note that not all algorithms (or concrete configurations of a specific algorithm)
+            support this behavior.
+
+            When specified as a string (in elkg or elkt) the individual options must be separated by ';,;'.
```
Comment (le-cds) [inline plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/Core.melk:403]: Why such a complicated separator?

Edit: This is explained somewhere in a Javadoc comment below. Personally, I'd prefer a simpler separator which is escaped whenever the actual character should appear in a string. I think a `;` should work well, shouldn't it? I don't think it appears in any spacing option?

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/IndividualSpacings.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2016 Kiel University and others.
+ * Copyright (c) 2016, 2020 Kiel University and others.
```
Comment (le-cds) [inline plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/IndividualSpacings.java:2]: Have the copyright years been updated on the other files as well?

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/IndividualSpacings.java [hunk 4]
```diff
@@ -88,5 +105,48 @@
         
         return result;
     }
+    
+    /** A (hopefully) unique separator that allows single occurrences of commas, colons, and semi-colons in-between. */
+    private static final String SERIALIZED_OPTION_SEPARATOR = ";,;";
+    
+    /**
+     * {@inheritDoc}
+     */
```
Comment (le-cds) [inline plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/IndividualSpacings.java:114]: Should be removed.

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/IndividualSpacings.java [hunk 5]
```diff
@@ -88,5 +105,48 @@
...
+    public String toString() {
+        final String serialized = getAllProperties().entrySet().stream()
+                .map(entry -> entry.getKey().getId() + ":" + entry.getValue().toString())
+                .collect(Collectors.joining(SERIALIZED_OPTION_SEPARATOR));
+        return serialized;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
```
Comment (le-cds) [inline plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/IndividualSpacings.java:125]: Should be removed.

File: plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/Core.melk (hunk 2 above)
Comment (uruuru) [reply plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/Core.melk:403]: It does. It's an allowed separator for margins and paddings, e.g.

**Pull request merged**